### PR TITLE
fix: update sst to latest

### PR DESCRIPTION
**Pull request by vasco-santos**
Fixes Deployment issues by finally updating to the LATEST sst and update aws cdk/sdk deps to be compatible. See [migration guide](https://docs.sst.dev/upgrade-guide) for details. In reality, the core changes apply right to [upgrade-guide#upgrade-to-v20](https://docs.sst.dev/upgrade-guide#upgrade-to-v20), given remaining items is things we do not use today

You can run to know what versions should be used:
```sh
➜  w3infra git:(fix/update-sst-to-v2) npx sst version
SST: v2.40.3
CDK: v2.124.0
Constructs: v10.3.0
```

❗ while the github PR diff seems massive, actually most is on the `package-lock`, and changes and mostly methodical and the same things all over the place, like changing import paths

A few notes:
- srcPath was sadly removed on sst, which makes us need to use the entire path
- updated `multiformats` + `@ipld/car` dependencies because of the big changes in package-lock there were mismatch versions causing type issues in Links
- updated `testcontainers`  given billing already had a new one and local tests were flakey because sometimes too many containers open take too long to stop/start

### CODE REVIEW

File: filecoin/test/compute-piece-cid.test.js [hunk 1]
```diff
@@ -14,20 +14,29 @@ const AWS_REGION = 'us-west-2'
...
 
   Object.assign(t.context, {
     s3Client: client,
     dbEndpoint,
-    dynamoClient
+    dynamoClient,
+    stop: async () => {
+      await s3Stop()
+      await dynamoStop()
+    }
```
Comment (vasco-santos) [inline filecoin/test/compute-piece-cid.test.js:32]: forcing stop to decrease likelihood of flakey tests

File: package.json [hunk 2]
```diff
@@ -12,17 +12,15 @@
     "console": "sst console",
     "lint": "tsc && eslint '**/*.js'",
     "clean": "rm -rf dist node_modules package-lock.json ./*/{.cache,dist,node_modules}",
-    "test": "npm test -w billing -w upload-api -w carpark -w replicator -w satnav -w ucan-invocation -w roundabout -w filecoin",
```
Comment (vasco-santos) [inline package.json:15]: we do not have that package anymore since https://github.com/web3-storage/w3infra/pull/315

File: sst.config.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
+import type { SSTConfig } from 'sst'
```
Comment (vasco-santos) [inline sst.config.ts:1]: basically `stacks/index.js` is now this file :)

File: test/helpers/deployment.js [hunk 4]
```diff
@@ -31,11 +42,15 @@ export const getAwsBucketClient = (region = getAwsRegion()) => new S3Client({
...
   if (process.env.SEED_APP_NAME) {
+    const stage = getStage()
     return `https://${stage}.up.web3.storage`
   }
 
   const require = createRequire(import.meta.url)
-  const testEnv = require('../../.test-env.json')
+  const testEnv = require(path.join(
+    process.cwd(),
+    '.sst/outputs.json'
```
Comment (vasco-santos) [inline test/helpers/deployment.js:52]: this was changed from new version, but of course too internal to be included in the migration guide... one of those "needle in a haystack" that took forever to get tests working with dynamic deployment of infra per PR stage

**Comment by olizilla** [comment]
you make it looks so easy! I thought this was gonna be much more of faff.

Do you know if sst will figure out that the stateful resources like buckets and tables are the same thing across this upgrade. Need to verify that it's not gonna try and delete and recreate them.

**Comment by vasco-santos** [comment]
> Do you know if sst will figure out that the stateful resources like buckets and tables are the same thing across this upgrade. Need to verify that it's not gonna try and delete and recreate them.

yeah, that is something I pointed out to @alanshaw . We need to see what the diff will be when doing the release with seed.run in staging to be totally sure. But I would say that it won't recreate, at least from the output I see on seed.run at the moment:

image

File: test/helpers/deployment.js [hunk 5]
```diff
@@ -1,18 +1,29 @@
...
+  if (stage) {
+    return stage
+  }
+
+  const f = fs.readFileSync(path.join(
+    process.cwd(),
+    '.sst/stage'
+  ))
+
+  return f.toString()
```
Comment (alanshaw) [inline test/helpers/deployment.js:21]:
```suggestion
  return fs.readFileSync(path.join(process.cwd(), '.sst', 'stage'), 'utf8')
```

File: test/helpers/deployment.js [hunk 6]
```diff
@@ -79,14 +102,18 @@ export const getSatnavBucketInfo = () => {
...
       Region: 'us-east-2'
     }
   }
 
   const require = createRequire(import.meta.url)
-  const testEnv = require('../../.test-env.json')
+  const testEnv = require(path.join(
+    process.cwd(),
+    '.sst/outputs.json'
+  ))
```
Comment (alanshaw) [inline test/helpers/deployment.js:116]: Pull out as a function?

**Pull request merged**